### PR TITLE
Fix MessageReader copy_from_slice panic

### LINKED ISSUE #244: TiKV paniced if using the latest master, with zero-copy feature.
I tested tikv with latest master grpc-rs, by specifying this in `Cargo.toml`:
```toml
[replace]
"grpcio:0.4.0" = { git = "https://github.com/pingcap/grpc-rs.git" }
```
But tikv paniced later. From the log the error happens in something about `grpc_slice`.
Please take a look. The log is attached.
[tikv.log](https://github.com/pingcap/grpc-rs/files/2500053/tikv.log)

### LINKED ISSUE #245: MessageReader panicked at `copy_from_slice`
```
2018/10/21 22:37:16.140 ERRO mod.rs:463: thread 'grpc-server-3' panicked 'assertion failed: `(left == right)`
  left: `11`,
 right: `40`: destination and source slices have different lengths' at "libcore/slice/mod.rs:1654"
stack backtrace:
   0:     0x7fba7319e54c - backtrace::backtrace::libunwind::trace::h502fbb448fc3a245
                        at /home/qupeng/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.2.3/src/backtrace/libunwind.rs:54
                         - backtrace::backtrace::trace::h3d8c31c6e03be1c9
                        at /home/qupeng/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.2.3/src/backtrace/mod.rs:70
   1:     0x7fba72cdd190 - tikv::util::set_exit_hook::{{closure}}::h2eec0714ac25aac9
                        at /home/qupeng/github.com/pingcap/tikv/target/release/build/backtrace-268b6593d69dacc9/out/capture.rs:79
   2:     0x7fba73317e1f - std::panicking::rust_panic_with_hook::h60a37fb79e459606
                        at libstd/panicking.rs:479
   3:     0x7fba73317c7e - std::panicking::continue_panic_fmt::hb542a58a3d52cb5c
                        at libstd/panicking.rs:390
   4:     0x7fba7331b0a5 - rust_begin_unwind
                        at libstd/panicking.rs:325
   5:     0x7fba73329c42 - core::panicking::panic_fmt::h68c841fe431ac371
                        at libcore/panicking.rs:77
   6:     0x7fba73163844 - core::slice::<impl [T]>::copy_from_slice::h81b1dc1f35bc6103
                        at /checkout/src/libcore/macros.rs:26
                         - <grpcio::call::MessageReader as std::io::Read>::read::haf8a78c48eace031
                        at /home/qupeng/github.com/pingcap/grpc-rs/src/call/mod.rs:171
   7:     0x7fba7299ffb0 - std::io::Read::read_exact::h43fe3fe4796f60f7
                        at /checkout/src/libstd/io/mod.rs:712
   8:     0x7fba73207621 - std::io::impls::<impl std::io::Read for &'a mut R>::read_exact::hbc1a16940b8e5221
                        at /checkout/src/libstd/io/impls.rs:43
                         - protobuf::buf_read_iter::BufReadIter::read_exact::he8aab5ba4e5a9574
                        at /home/qupeng/.cargo/registry/src/github.com-1ecc6299db9ec823/protobuf-2.0.4/src/buf_read_iter.rs:284
                         - protobuf::stream::CodedInputStream::read::hc95606664f4d13f0
                        at /home/qupeng/.cargo/registry/src/github.com-1ecc6299db9ec823/protobuf-2.0.4/src/stream.rs:184
   9:     0x7fba73212a44 - protobuf::stream::CodedInputStream::read_raw_bytes_into::h50461e61aa6df85d
                        at /home/qupeng/.cargo/registry/src/github.com-1ecc6299db9ec823/protobuf-2.0.4/src/stream.rs:633
                         - protobuf::stream::CodedInputStream::read_bytes_into::hb4e83f02d2b07d0b
                        at /home/qupeng/.cargo/registry/src/github.com-1ecc6299db9ec823/protobuf-2.0.4/src/stream.rs:669
  10:     0x7fba73121d3c - <kvproto::raft_serverpb::RaftMessage as protobuf::core::Message>::merge_from::h0e40227ddc36e857
                        at /home/qupeng/.cargo/registry/src/github.com-1ecc6299db9ec823/protobuf-2.0.4/src/rt.rs:751
  11:     0x7fba72b66a7a - grpcio::codec::pb_codec::de::h9cbd1884cc25f9c1
                        at /home/qupeng/github.com/pingcap/grpc-rs/src/codec.rs:54
  12:     0x7fba72c50b82 - core::ops::function::FnOnce::call_once::h6fbbe2d724a41169
```

### PR DESCRIPTION
**Pull request by overvenus**
`copy_from_slice` requires dest and source have the same length.

Close #244 
Close #245

### CODE REVIEW

**Comment by hicqu** [comment]
LGTM.

**Comment by overvenus** [comment]
PTAL @BusyJay

File: src/call/mod.rs [hunk 1]
```diff
@@ -14,9 +14,10 @@
 pub mod client;
 pub mod server;
 
+use std::cmp;
 use std::io::{self, BufRead, ErrorKind, Read};
 use std::sync::Arc;
-use std::{cmp, mem, ptr, slice, usize};
+use std::{mem, ptr, slice, usize};
```
Comment (ice1000) [inline src/call/mod.rs]: Why?

Comment (ice1000) [reply src/call/mod.rs]: GitHub is only showing 4 lines which makes my comment misleading.

Look at this pic:
image

**Comment by overvenus** [comment]
PTAL @ice1000 , thanks!

**Pull request merged**